cartography/intel/aws/ec2/auto_scaling_groups.py

```diff
@@ -26,88 +26,94 @@ def get_ec2_auto_scaling_groups(boto3_session: boto3.session.Session, region: st
+        group.maxsize = ag.MaxSize, group.minsize = ag.MinSize, group.defaultcooldown = ag.DefaultCooldown,
+        group.desiredcapacity = ag.DesiredCapacity, group.healthchecktype = ag.HealthCheckType,
+        group.healthcheckgraceperiod = ag.HealthCheckGracePeriod, group.status = ag.Status,
+        group.newinstancesprotectedfromscalein = ag.NewInstancesProtectedFromScaleIn,
+        group.maxinstancelifetime = ag.MaxInstanceLifetime, group.capacityrebalance = ag.CapacityRebalance,
+        group.region={Region}
+        WITH group
+        MATCH (aa:AWSAccount{id: {AWS_ACCOUNT_ID}})
+        MERGE (aa)-[r:RESOURCE]->(group)
+        ON CREATE SET r.firstseen = timestamp()
+        SET r.lastupdated = {update_tag}
     """
 
     ingest_vpc = """
-    MERGE (subnet:EC2Subnet{subnetid: {SubnetId}})
-    ON CREATE SET subnet.firstseen = timestamp()
-    SET subnet.lastupdated = {update_tag}
-    WITH subnet
-    MATCH (group:AutoScalingGroup{arn: {GROUPARN}})
-    MERGE (subnet)<-[r:VPC_IDENTIFIER]-(group)
-    ON CREATE SET r.firstseen = timestamp()
-    SET r.lastupdated = {update_tag}
+    UNWIND {vpc_list} as vpc
+        MERGE (subnet:EC2Subnet{subnetid: vpc.SubnetId})
+        ON CREATE SET subnet.firstseen = timestamp()
+        SET subnet.lastupdated = {update_tag}
+        WITH subnet
+        MATCH (group:AutoScalingGroup{arn: {GROUPARN}})
+        MERGE (subnet)<-[r:VPC_IDENTIFIER]-(group)
+        ON CREATE SET r.firstseen = timestamp()
+        SET r.lastupdated = {update_tag}
     """
 
     ingest_instance = """
-    MERGE (instance:Instance:EC2Instance{id: {InstanceId}})
-    ON CREATE SET instance.firstseen = timestamp()
-    SET instance.instanceid = {InstanceId}, instance.lastupdated = {update_tag}, instance.region={Region}
-    WITH instance
-    MATCH (group:AutoScalingGroup{arn: {GROUPARN}})
-    MERGE (instance)-[r:MEMBER_AUTO_SCALE_GROUP]->(group)
-    ON CREATE SET r.firstseen = timestamp()
-    SET r.lastupdated = {update_tag}
-    WITH instance
-    MATCH (aa:AWSAccount{id: {AWS_ACCOUNT_ID}})
-    MERGE (aa)-[r:RESOURCE]->(instance)
-    ON CREATE SET r.firstseen = timestamp()
-    SET r.lastupdated = {update_tag}
+    UNWIND {instances_list} as i
+        MERGE (instance:Instance:EC2Instance{id: i.InstanceId})
+        ON CREATE SET instance.firstseen = timestamp()
+        SET instance.lastupdated = {update_tag}, instance.region={Region}
+        WITH instance
+        MATCH (group:AutoScalingGroup{arn: {GROUPARN}})
+        MERGE (instance)-[r:MEMBER_AUTO_SCALE_GROUP]->(group)
+        ON CREATE SET r.firstseen = timestamp()
+        SET r.lastupdated = {update_tag}
+        WITH instance
+        MATCH (aa:AWSAccount{id: {AWS_ACCOUNT_ID}})
+        MERGE (aa)-[r:RESOURCE]->(instance)
+        ON CREATE SET r.firstseen = timestamp()
+        SET r.lastupdated = {update_tag}
     """
 
+    # neo4j does not accept datetime objects and values. This loop is used to convert
```

Comment:
Nit: technically not correct anymore as of 3.5 iirc, but it's fine to keep as `str()`.

Reply:
Ah I see. Okay, I'll just remove that comment.